pjsip/src/pjsip/sip_transaction.c

```diff
@@ -2042,6 +2042,38 @@ static void transport_callback(void *token, pjsip_tx_data *tdata,
      */
     pj_grp_lock_acquire(tsx->grp_lock);
     tsx->transport_flag &= ~(TSX_HAS_PENDING_TRANSPORT);
+
+    if (sent > 0) {
+	/* Pending destroy? */
+	if (tsx->transport_flag & TSX_HAS_PENDING_DESTROY) {
+	    tsx_set_state( tsx, PJSIP_TSX_STATE_DESTROYED,
+			   PJSIP_EVENT_UNKNOWN, NULL, 0 );
+	    pj_grp_lock_release(tsx->grp_lock);
+	    return;
+	}
+
+	/* Need to transmit a message? */
+	if (tsx->transport_flag & TSX_HAS_PENDING_SEND) {
+	    tsx->transport_flag &= ~(TSX_HAS_PENDING_SEND);
+	    tsx_send_msg(tsx, tsx->last_tx);
```

Comment:
perhaps better to have some check/protection?
```
if (tsx->last_tx && tsx->last_tx != tdata)
  tsx_send_msg();
```

Reply:
The block is actually identical to the one in `send_msg_callback()`. So should I add the check in both? Or is there any reason why the check is only applicable for this callback but not the other?

Reply:
ah wasn't aware of that, as so far `send_msg_callback()` has been proven to be safe, I think this callback should be safe too without the checks.